### PR TITLE
VACMS-4609 Replace migration calls to time() with custom callable.

### PR DESCRIPTION
**Pull request by swirtSJW**
## Description

See #4609. 

## QA steps

As user _uid_ with _user_role_
1. Do this
1. Then that
1. Then validate Acceptance Criteria from issue
- [ ] This
- [ ] That
- [ ] The other thing

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [ ] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by swirtSJW** [comment]
Ran locally no issues.  
image

**Comment by swirtSJW** [comment]
Darn.   Ignore that test.  I just looked and somehow despite a destroy I still have the old version 
image

**Comment by swirtSJW** [comment]
Tested for real this time  ;)

image

Confirmed we are all good.



image

Times are coming in as expected.
image
image

**Comment by mchelen-gov** [comment]
Do we know why the migration calls to `time()` are passing it any arguments?

File: docroot/modules/custom/va_gov_migrate/va_gov_migrate.module [hunk 1]
```diff
@@ -65,3 +65,20 @@ function va_gov_migrate_help($route_name, RouteMatchInterface $route_match) {
...
+ * Reference https://bugs.php.net/bug.php?id=78509
+ *
+ * @param mixed $arg
+ *   An argument from the process plugin pipeline.  It is not used.
+ *
+ * @return int
+ *   The current time measured in the number of seconds
+ */
+function va_gov_migrate_time($arg) {
+  // Process pipeline passes the return of the previous process.
```
Comment (mchelen-gov) [inline docroot/modules/custom/va_gov_migrate/va_gov_migrate.module:81]: is this a behavior of drupal or drupal module?

Comment (ElijahLynn) [reply docroot/modules/custom/va_gov_migrate/va_gov_migrate.module:81]: per Slack from @swirtSJW:
> process plugins in migrations run in a chain, one passes results to the next, so they always pass something... even if they pass null.   Which in this case was passing NULL to time() which in the past has never been an issue.   It used to be no-matter what you passed to time would do its thing and return the time.... now it's gone all hardened and if you pass something to it , it returns NULL.

Comment (swirtSJW) [reply docroot/modules/custom/va_gov_migrate/va_gov_migrate.module:81]: Its the way core migrate process plugins chain.  By design they have to pass something.

**Comment by ElijahLynn** [comment]
> I deployed this PR to PROD after tests passed on STAGING. Not sure why yet, but getting same error on PROD in logs and on Periodic Job. I suppose I should have tested on STAGING Periodic Job first.

I still see a bunch of `callable: time` in the latest master branch, in the config/sync/ files. The changes in this PR were all in config/install. Testing locally now.

**Comment by ElijahLynn** [comment]
> I still see a bunch of `callable: time` in the latest master branch, in the config/sync/ files. The changes in this PR were all in config/install. Testing locally now.

This should be fixed by @swirtSJW in https://github.com/department-of-veterans-affairs/va.gov-cms/pull/4611 now.

**Pull request merged**